### PR TITLE
Add steps for custom certs

### PR DESCRIPTION
**Pull request by timuthy**
**What this PR does / why we need it**:
This PR adds the necessary steps to request certificates for custom domains.

**Release note**:

```improvement operator
The developer documentation now contains information on how to request certificates for custom domains.
```

### CODE REVIEW

File: website/documentation/050-tutorials/content/howto/x509_certificates/_index.md [hunk 1]
```diff
@@ -82,6 +82,44 @@ status:
   state: Ready
 ```
 
+## Custom Domains
+If you want to request certificates for domains other then any subdomain of `shoot.spec.dns.domain`, the following configuration is required:
+
+### DNS provider
+In order to issue certificates for a custom domain you need to specify a DNS provider which is permitted to create DNS records for subdomains of your requested domain in the certificate. For example, if you request a certificate for `host.example.com` you DNS provider must be capable of managing subdomains of `host.example.com`.
```
Comment (g-pavlov) [inline website/documentation/050-tutorials/content/howto/x509_certificates/_index.md]:
```suggestion
In order to issue certificates for a custom domain you need to specify a DNS provider which is permitted to create DNS records for subdomains of your requested domain in the certificate. For example, if you request a certificate for `host.example.com` your DNS provider must be capable of managing subdomains of `host.example.com`.
```

File: website/documentation/050-tutorials/content/howto/x509_certificates/_index.md [hunk 2]
```diff
@@ -82,6 +82,44 @@ status:
...
+
+DNS providers are specified in the shoot manifest:
+
+```yaml
+kind: Shoot
+...
+spec:
+  dns:
+    providers:
+    - type: aws-route53 # consult https://github.com/gardener/external-dns-management#using-the-dns-controller-manager for possible values
```
Comment (g-pavlov) [inline website/documentation/050-tutorials/content/howto/x509_certificates/_index.md]: Do you refer to the DNS provisioning controllers group?
```suggestion
    - type: aws-route53 # consult the DNS provisioning controllers group (dnscontrollers) in https://github.com/gardener/external-dns-management#using-the-dns-controller-manager for possible values
```

File: website/documentation/050-tutorials/content/howto/x509_certificates/_index.md [hunk 3]
```diff
@@ -82,6 +82,44 @@ status:
...
+      secretName: provider-example-com # contains credentials for service account, see https://github.com/gardener/external-dns-management/tree/master/examples
```
Comment (g-pavlov) [inline website/documentation/050-tutorials/content/howto/x509_certificates/_index.md]:
```suggestion
      secretName: provider-example-com # contains credentials for service account, see any 20-secret-<provider>-credentials.yaml in https://github.com/gardener/external-dns-management/tree/master/examples
```

**Comment by timuthy** [comment]
Thanks for the review @g-pavlov.

**Pull request merged**